src/django/api/views.py

```diff
@@ -1057,14 +1057,86 @@ def split(self, request, pk=None):
         except Facility.DoesNotExist:
             raise NotFound()
 
-    @action(detail=True, methods=['GET', 'POST'],
+    @action(detail=True, methods=['POST'],
             permission_classes=(IsRegisteredAndConfirmed,))
     @transaction.atomic
     def promote(self, request, pk=None):
         if not request.user.is_superuser:
             raise PermissionDenied()
+        match_id = request.data.get('match_id')
+
+        if match_id is None:
+            raise BadRequestException('Missing required param match_id')
+
+        try:
+            facility = Facility \
+                .objects \
+                .get(pk=pk)
+
+            match = FacilityMatch \
+                .objects \
+                .get(pk=match_id)
+
```

Comment:
Consider raising `BadRequestException` in these cases
- `match.facility != facility`
We should only allow promotion if the match already points to the facility
- `facility.created_from == match.facility_list_item`
We don't need to promote if the match is already the "winner"

Reply:
Good idea -- I added these checks and also tests for them in 4c8f50d7ab258527c7915cdd825cbfccae7ccf5f